### PR TITLE
[ENG-1130] Add review page

### PR DESCRIPTION
**Pull request by fabmiz**
- Ticket: [ENG-1130](https://openscience.atlassian.net/browse/ENG-1130)
- Feature flag: n/a

## Purpose

Add review page

## Summary of Changes

- Use `Registries::ReviewFormRenderer`
- Add navigation tests
- Remove `ember-router-action-helper`
    - Use route controller since we expect to add state over time.
    - Move route actions to controller.

## Side Effects
n/a
## QA Notes
n/a

### CODE REVIEW

File: lib/registries/addon/drafts/draft/controller.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    @action
+    updateRoute(headingText: string) {
+        if (this.page) {
+            const pageSlug = getPageParam(this.pageIndex, headingText);
+            this.transitionToRoute('drafts.draft', this.draftId, pageSlug);
+        }
+    }
+
+    @action
+    error() {
```
Comment (jamescdavis) [inline lib/registries/addon/drafts/draft/controller.ts]:
```suggestion
    onPageNotFound() {
```

File: lib/registries/addon/drafts/draft/template.hbs [hunk 2]
```diff
@@ -1,55 +1,93 @@
...
+    <div local-class='ContentBackground Loading'>
+        <LoadingIndicator @dark={{true}} />
+    </div>
+{{else}}
+    <Registries::DraftRegistrationManager
+        @currentPage={{this.pageIndex}}
+        @draftRegistration={{this.draftRegistration}}
+        @inReview={{this.inReview}}
+        @updateRoute={{action this.updateRoute}}
+        @onPageNotFound={{action this.error}}
```
Comment (jamescdavis) [inline lib/registries/addon/drafts/draft/template.hbs]: don't forget to rename

File: lib/registries/addon/drafts/draft/route.ts [hunk 3]
```diff
@@ -2,13 +2,20 @@ import { action } from '@ember-decorators/object';
...
 import requireAuth from 'ember-osf-web/decorators/require-auth';
+import DraftRegistration from 'ember-osf-web/models/draft-registration';
 import Analytics from 'ember-osf-web/services/analytics';
 
-import { getPageIndex, getPageParam } from 'ember-osf-web/utils/page-param';
+import { getPageIndex } from 'ember-osf-web/utils/page-param';
+
+export interface DraftRouteModel {
+    taskInstance: TaskInstance<DraftRegistration>;
+    pageIndex: number|undefined;
```
Comment (jamescdavis) [inline lib/registries/addon/drafts/draft/route.ts]:
```suggestion
    pageIndex?: number;
```

Comment (fabmiz) [reply lib/registries/addon/drafts/draft/route.ts]: will do on the next register PR

**Pull request merged**